### PR TITLE
base - key int instead of string

### LINKED ISSUE #14242: Bug in the error handling module in Bitfinex
Hi team, 
During order placement on Bitfinex, I'm seeing an issue in the module. The original error seems to be a transient network related error but another exception arises while handling that error. Attaching the stack trace details below.

- OS: Mac OS
- Programming Lang: Python
- CCXT version: 1.90.24

Stack trace:

Traceback (most recent call last):
  File '/home/cleon/qa/cleoninterface/ExchangeModule/exchanges/ExchangeWrapperUtils.py', line 14, in create_order_on_exchange
    order = exchange_instance.create_order(symbol, type, side, amount, price=price, params=params)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/bitfinex.py', line 986, in create_order
    response = self.privatePostOrderNew(self.extend(request, params))
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 497, in inner
    return entry(_self, **inner_kwargs)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 2616, in request
    return self.fetch2(path, api, method, params, headers, body, config, context)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 2613, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 644, in fetch
    raise NetworkError(details) from e
ccxt.base.errors.NetworkError: bitfinex POST https://api.bitfinex.com/v1/order/new

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 617, in fetch
    response.raise_for_status()
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/requests/models.py', line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://api.bitfinex.com/v1/order/new

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File '/home/cleon/qa/cleoninterface/ExchangeModule/exchanges/ExchangeWrapperUtils.py', line 14, in create_order_on_exchange
    order = exchange_instance.create_order(symbol, type, side, amount, price=price, params=params)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/bitfinex.py', line 986, in create_order
    response = self.privatePostOrderNew(self.extend(request, params))
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 497, in inner
    return entry(_self, **inner_kwargs)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 2616, in request
    return self.fetch2(path, api, method, params, headers, body, config, context)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 2613, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 633, in fetch
    skip_further_error_handling = self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/bitfinex.py', line 1495, in handle_errors
    firstChar = self.safe_string(body, 0)
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 704, in safe_string
    return str(dictionary[key]) if Exchange.key_exists(dictionary, key) else default_value
  File '/home/cleon/qa/venv2/lib/python3.7/site-packages/ccxt/base/exchange.py', line 688, in key_exists
    if key in dictionary:
TypeError: 'in <string>' requires string as left operand, not int

----- -----

From my initial investigation, the error seems to be in the ccxt/bitfinex.py module in the `handle_errors` method on line `firstChar = self.safe_string(body, 0)`. The `safe_string` function expects the second argument to be a string.

Request you to look into this at the earliest possible.

### PR DESCRIPTION
**Pull request by ttodua**
fix #14242

### CODE REVIEW

File: python/ccxt/base/exchange.py [hunk 1]
```diff
@@ -688,7 +688,8 @@ def key_exists(dictionary, key):
                 return dictionary[key] is not None
             else:
                 return False
-        if key in dictionary:
+        # check for string type, to avoid  `TypeError: 'in ' requires string as left operand, not int`, i.e. safeString(dict, 0)
+        if isinstance(key, str) and key in dictionary:
```
Comment (carlosmiei) [inline python/ccxt/base/exchange.py:692]: @ttodua Can you please elaborate a bit more on this? I'm not getting that error when using integer keys

Comment (ttodua) [reply python/ccxt/base/exchange.py:692]: I had incorrect impression. the linked issue seems to be having different problem - not `integer in list`, but `integer in string` check. The error log was:
```
File 'ccxt/bitfinex.py', line 1495, in handle_errors
    firstChar = self.safe_string(body, 0)
File 'ccxt/base/exchange.py', line 704, in safe_string
    return str(dictionary[key]) if Exchange.key_exists(dictionary, key) else default_value
File 'ccxt/base/exchange.py', line 688, in key_exists
    if key in dictionary:
TypeError: 'in ' requires string as left operand, not int
```

so there `body` seems to be `string` already. how it should be handled effectively ?  that might need to fix globally, not for just `bitfinex`...

Comment (kroitor) [reply python/ccxt/base/exchange.py:692]: @ttodua 

> how it should be handled effectively ?

```JavaScript
if (typeof body === 'string') {
    // body is a string
} else {
    // not a string
}
```

**Comment by kroitor** [comment]
> fix #14242

That issue seems to be already closed, and tbh i'm not entirely sure how `key in dictionary` fixes it, I think we can just add a guard in bitfinex's handleErrors and that would work, no (do we really need to change the base class here) ?

**Pull request closed without merging**